### PR TITLE
add context to syntax errors

### PR DESCRIPTION
**Pull request by fmoor**
fixes https://github.com/edgedb/edgedb-go/issues/116

A query snip-it with arrow and hint is introduced for syntax errors.

## Caveats 
This implementation is very crude and likely to work poorly with characters outside of the ascii range because bytes are construed as characters. It also replaces tabs with spaces in the query to avoid misalignment when printed.

There doesn't seem to be a library in go similar to rust's [codespan](https://github.com/brendanzab/codespan) so the only option seems to be to roll our own.

## Example
`SELECT (foo (((1 2) 3)) 4)` produces
```
edgedb.EdgeQLSyntaxError: Unexpected token: <Token ICONST "2">
query:1:18

SELECT (foo (((1 2) 3)) 4)
                 ^ It appears that a ',' is missing in a tuple before '2'
```

### CODE REVIEW

**Comment by fmoor** [comment]
If it seems worth while I could extend this to work with arbitrary unicode instead of just the ascii range.

**Comment by fmoor** [comment]
This change is based on https://github.com/edgedb/edgedb-go/pull/119 and is WIP until the upstream branch is merged.

**Comment by 1st1** [comment]
> If it seems worth while I could extend this to work with arbitrary unicode instead of just the ascii range.

Yes, we should do that. The updated error messages is a major improvement.

**Comment by 1st1** [comment]
I'll wait until the review until https://github.com/edgedb/edgedb-go/pull/119 is merged.

File: error.go
Comment (1st1) [inline error.go:86]: Well, all kinds of errors can have position information in them. IMO we should instead check for that, and then render a nice message for all kinds of errors. E.g. a query might be correct syntactically, but have an invalid type reference; the server would return the exact spot of the problem in the query.

Comment (fmoor) [reply error.go]: changed

**Comment by fmoor** [comment]
> I could extend this to work with arbitrary unicode instead of just the ascii range.

This is a little better now, but it turns out that character width, even in a fixed width font, is font dependent. `世界` is 2 code points but each character takes up 2 columns in my terminal font. I haven't found a nice way of determining the number of columns a grapheme cluster typically consumes.

**Review by 1st1 — APPROVED** [review]

**Pull request merged**
